test/e2e/dashboard_test.go

```diff
@@ -0,0 +1,82 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	configv1 "github.com/openshift/api/config/v1"
+	monitoringdashboard "github.com/openshift/cluster-ingress-operator/pkg/operator/controller/monitoring-dashboard"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"time"
+)
+
+func TestDashboardCreation(t *testing.T) {
+	t.Parallel()
+
+	infraConfig := &configv1.Infrastructure{}
+	if err := kclient.Get(context.TODO(), types.NamespacedName{Name: "cluster"}, infraConfig); err != nil {
+		t.Fatalf("failed to get infraConfig: %v", err)
+	}
+
+	dashboardCM := &corev1.ConfigMap{}
+	if err := kclient.Get(context.TODO(), monitoringdashboard.ConfigMapName(), dashboardCM); err != nil {
+		if errors.IsNotFound(err) && infraConfig.Status.ControlPlaneTopology == configv1.ExternalTopologyMode {
+			// Dashboard is not created when external topology is externel
+			return
+		}
+		t.Fatalf("failed to get dashboard configmap: %v", err)
+	}
+
+	initialData := dashboardCM.Data
+
+	// Change dashboard in configmap and check for update from the operator
+	dashboardCM.Data = map[string]string{
+		"dashboard.json": "",
+	}
+	if err := kclient.Update(context.TODO(), dashboardCM); err != nil {
+		t.Fatalf("failed to update dashboard configmap: %v", err)
+	}
+	err := wait.PollImmediate(1*time.Second, 1*time.Minute, func() (bool, error) {
+		if err := kclient.Get(context.TODO(), monitoringdashboard.ConfigMapName(), dashboardCM); err != nil {
+			t.Logf("failed to get configmap: %v, retrying...", err)
+			return false, nil
+		}
+		dashboard, ok := dashboardCM.Data["dashboard.json"]
+		if !(ok && len(dashboard) > 0) {
+			t.Logf("Controller did not modify the ConfigMap back to its original state, retrying...")
+			return false, nil
+		}
+		return true, nil
+	})
```

Comment:
I think this could be a little simpler: 

```suggestion
	err := wait.PollImmediate(1*time.Second, 1*time.Minute, func() (bool, error) {
		if err := kclient.Get(context.TODO(), monitoringdashboard.ConfigMapName(), dashboardCM); err != nil {
			t.Logf("Retrying due to error: %v", err)
			return false, nil
		}
		if dashboard, ok := dashboardCM.Data["dashboard.json"]; ok && dashboard != "" {
			return true, nil
		}
		t.Log("Awaiting dashboard update...")
		return false, nil
	})
```

Reply:
Thanks!
Github did not allow me to directly commit your changes because of changes to add the "dashbord.json" constant but I added them to the last commit.

Reply:
No problem.